source/dataset.py

```diff
@@ -1,35 +1,51 @@
-import numpy as np
-
+from datetime import datetime
 
-from torch.utils.data import Dataset
-from torch import from_numpy
+import numpy as np
 from sunpy.map import Map
-from datetime import datetime
+from torch import from_numpy
+from torch.utils.data import Dataset
 
-from source.data_utils import get_patch, get_array_radius, map_prep, scale_rotate
+from source.data_utils import get_patches, get_array_radius, map_prep, scale_rotate
 
 
 class FitsFileDataset(Dataset):
     """
     Construct a dataset of patches from a fits file
     """
-    def __init__(self, file, size, norm, instrument, rescale, upscale_factor):
 
-        map = map_prep(file, instrument)
-        map.data[:] = map.data[:]/norm
+    def __init__(self, file, size, config_data):
+
+        self.instrument = config_data['instrument']
+        self.norm = config_data['data']['normalisation']
+        self.rescale = config_data['data']['rescale']
+        self.upscale_factor = config_data['net']['upscale_factor']
+        self.padding = config_data['data']['padding']
+        self.loss = config_data['loss']
+        self.add_noise = config_data['cli']['add_noise']
+
+        amap = map_prep(file, self.instrument)
+        amap.data[:] = amap.data[:] / self.norm
 
         # Detecting need for rescale
-        if rescale and np.abs(1 - (map.meta['cdelt1']/0.504273)/upscale_factor) > 0.01:
-            map = scale_rotate(map, target_factor=upscale_factor)
+        if self.rescale and np.abs(1 - (amap.meta['cdelt1']/0.504273) / self.upscale_factor) > 0.01:
```

Comment:
This is a bit of a magic number not totally sure what we are doing here I would suggest 
1. create a constant at the top for the file and reference that or 
2. since this is a class put it in the `__init__` with a default value and document it